src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1516,6 +1519,15 @@ namespace winrt::TerminalApp::implementation
                     text = item.Path();
                 }
             }
+            //The following removes all tab characters when pasting from clipboard.
+            if (removeTabs)
+            {
+                hstring::value_type* temporary = new hstring::value_type[text.size()+1];
+                wcscpy_s(temporary, text.size() + 1, text.c_str());
+                auto endIt = std::remove(temporary, temporary + text.size() + 1, L'\t');
+                text = hstring(temporary, static_cast<winrt::hstring::size_type>(endIt - temporary));
+                delete[] temporary;
```

Comment:
If you don't want to use `new`, `delete`, and `wcscpy_s`, you could do it like this:
```c++
std::wstring temporary;
temporary.reserve(text.size());
std::copy_if(text.cbegin(), text.cend(), std::back_inserter(temporary),
             [](const wchar_t c) { return c != '\t'; });
text = temporary;
```

Reply:
I was actually just about to block this over the exact same thing. We should be avoiding raw `new`/`delete`'s as much as possible in new code like this.

Reply:
Fixed! Wasn't aware that wstring would work when hstring wouldn't, this solution is better. The only thing we changed was we did c != L'\t' instead, since I believe that's needed for a wchar.

Note: I really appreciate how responsive all of you are. It really makes contributing a lot easier!